File: src/main/java-graphql/io/github/graphql/model/ReviewDismissalAllowanceTO.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer;
import java.util.StringJoiner;

@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class ReviewDismissalAllowanceTO implements java.io.Serializable, NodeTO {

    private ReviewDismissalAllowanceActorTO actor;
    private BranchProtectionRuleTO branchProtectionRule;
    @javax.validation.constraints.NotNull
    private String id;

    public ReviewDismissalAllowanceTO() {
    }

    public ReviewDismissalAllowanceTO(ReviewDismissalAllowanceActorTO actor, BranchProtectionRuleTO branchProtectionRule, String id) {
        this.actor = actor;
        this.branchProtectionRule = branchProtectionRule;
        this.id = id;
    }

    public ReviewDismissalAllowanceActorTO getActor() {
        return actor;
    }

    public BranchProtectionRuleTO getBranchProtectionRule() {
        return branchProtectionRule;
    }

    public String getId() {
        return id;
    }


    @Override
    public String toString() {
        StringJoiner joiner = new StringJoiner(", ", "{ ", " }");
        if (actor != null) {
            joiner.add("actor: " + GraphQLRequestSerializer.getEntry(actor));
        }
        if (branchProtectionRule != null) {
            joiner.add("branchProtectionRule: " + GraphQLRequestSerializer.getEntry(branchProtectionRule));
        }
        if (id != null) {
            joiner.add("id: " + GraphQLRequestSerializer.getEntry(id));
        }
        return joiner.toString();
    }

    public static ReviewDismissalAllowanceTO.Builder builder() {
        return new ReviewDismissalAllowanceTO.Builder();
    }

    public static class Builder {

        private ReviewDismissalAllowanceActorTO actor;
        private BranchProtectionRuleTO branchProtectionRule;
        private String id;

        public Builder() {
        }

        public Builder setActor(ReviewDismissalAllowanceActorTO actor) {
            this.actor = actor;
            return this;
        }

        public Builder setBranchProtectionRule(BranchProtectionRuleTO branchProtectionRule) {
            this.branchProtectionRule = branchProtectionRule;
            return this;
        }

        public Builder setId(String id) {
            this.id = id;
            return this;
        }


        public ReviewDismissalAllowanceTO build() {
            return new ReviewDismissalAllowanceTO(actor, branchProtectionRule, id);
        }

    }
}

File: src/main/graphql/io/github/graphql/model/ReviewDismissalAllowanceTO.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-05T19:09:00+0800"
)
open class ReviewDismissalAllowanceTO(
    val actor: ReviewDismissalAllowanceActorTO?,
    val branchProtectionRule: BranchProtectionRuleTO?,
    override
    val id: String
) : NodeTO {

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        if (actor != null) {
            joiner.add("actor: " + GraphQLRequestSerializer.getEntry(actor))
        }
        if (branchProtectionRule != null) {
            joiner.add("branchProtectionRule: " + GraphQLRequestSerializer.getEntry(branchProtectionRule))
        }
        joiner.add("id: " + GraphQLRequestSerializer.getEntry(id))
        return joiner.toString()
    }
}
